code/modules/power/apc.dm

```diff
@@ -220,6 +220,12 @@
 	..()
 	GLOB.apcs_list += src
 
+	var/area/local_area = get_area(src)
+	if (local_area)
+		if (local_area.apc)
+			WARNING("Duplicate APC created at [AREACOORD(src)]")
+		local_area.apc = src
```

Comment:
What happens if an apc moves?

Reply:
I suppose that can happen near a Singularity? I can listen for COMSIG_ENTER_AREA and COMSIG_EXIT_AREA if need be, but I have no idea if the existing behavior is coherent there

Reply:
@LemonInTheDark APCs moving happens so infrequently that we should just completely disconnect them when they do. APC code is hacked together enough as is from what I know, we don't even support two in the same area.

Reply:
that would be fine. just so long as the refs don't hang

Reply:
It's cleaned up in Destroy, which is also where the reference in the GLOB apc list is cleared, so the lifecycle should remain unchanged

A duplicate APC will clear an area's existing reference to an APC (and print a warning), but removing a reference early shouldn't cause it to hang around longer than clearing the reference later